synapse/storage/database.py

```diff
@@ -536,6 +538,31 @@ def new_transaction(
             **kwargs
         """
 
+        # Robustness check: ensure that none of the arguments are generators, since that
+        # will fail if we have to repeat the transaction.
+        # For now, we just log an error, and hope that it works on the first attempt.
```

Comment:
Should we update the docstring of this method to specifically call out not using generators?